srcalc/src/main/java/gov/va/med/srcalc/vista/RpcVistaPatientDao.java

```diff
@@ -160,56 +166,81 @@ private void parseRecentVitalResults(final Patient patient, final List<String> v
     	// Each entry comes with an accompanying date and time.
     	final String[] heightLineTokens = vitalResults.get(5).split(SPLIT_REGEX);
     	final int feet = Integer.parseInt(heightLineTokens[2]);
-    	patient.setHeight((feet * 12) + Integer.parseInt(heightLineTokens[4]));
-    	patient.setHeightDate(dateFormat.parse(heightLineTokens[1]));
+    	patient.setHeight(new RetrievedValue(
+    	        (feet * 12.0) + Double.parseDouble(heightLineTokens[4]),
+    	        dateFormat.parse(heightLineTokens[1]),
+    	        HEIGHT_UNITS));
     	final String[] weightLineTokens = vitalResults.get(6).split(SPLIT_REGEX);
-    	patient.setWeight(Double.parseDouble(weightLineTokens[2]));
-    	patient.setWeightDate(dateFormat.parse(weightLineTokens[1]));
+    	patient.setWeight(new RetrievedValue(
+    	        Double.parseDouble(weightLineTokens[2]),
+    	        dateFormat.parse(weightLineTokens[1]),
+    	        WEIGHT_UNITS));
     	final String[] bmiLineTokens = vitalResults.get(7).split(SPLIT_REGEX);
     	// The BMI value is the second to last token on its line
-    	patient.setBmi(Double.parseDouble(bmiLineTokens[bmiLineTokens.length-2]));
-    	patient.setBmiDate(patient.getWeightDate());
+    	patient.setBmi(new RetrievedValue(
+    	    Double.parseDouble(bmiLineTokens[bmiLineTokens.length-2]),
+    	    patient.getWeight().getMeasureDate(),
+    	    ""));
+    }
+    
+    private void retrieveLabs(final int dfn, final Patient patient) throws ParseException
+    {
+        final LabRetrievalEnum[] enumValues = LabRetrievalEnum.values();
+        for(final LabRetrievalEnum labNameList: enumValues)
+        {
+            final String rpcResultString = fProcedureCaller.doRetrieveLabs(
+                    fDuz,
+                    String.valueOf(dfn),
+                    labNameList.getPossibleLabNames());
+            // If the resultString is a success, add it to the patient's lab data.
+            // Else, we don't need to do anything.
+            if(!rpcResultString.isEmpty())
+            {
+                final String[] rpcSplit = rpcResultString.split("\\^");
+                final double labValue = Double.parseDouble(rpcSplit[1]);
+                final SimpleDateFormat format = new SimpleDateFormat("MM/dd/yyyy@HH:mm:ss");
+                patient.getLabs().put(labNameList.name(),
+                        new RetrievedValue(labValue, format.parse(rpcSplit[2]), rpcSplit[3]));
+            }
+        }
+    }
+    
+    @Override
+    public SaveNoteCode saveRiskCalculationNote(final int patientDfn, final String electronicSignature,
```

Comment:
I formatted these because the formatting was really out of whack when I looked at it in Eclipse.